### DISCUSSION TITLE
error trying to connect: invalid URL, scheme is not http

### DISCUSSION THREAD

**Discussion by tobymurray** [open-telemetry/opentelemetry-rust#1198] [Q&A] [answered]
I'm new to this entire ecosystem and I'm trying to use this crate with Honeycomb. My code looks like:

```
    let honeycomb_exporter = opentelemetry_otlp::new_exporter()
        .http()
        .with_headers(headers)
        .with_endpoint("https://api.honeycomb.io/v1/traces");

    let tracer = opentelemetry_otlp::new_pipeline()
        .tracing()
        .with_exporter(honeycomb_exporter)
        .install_batch(opentelemetry::runtime::Tokio)
        .unwrap();
```

Cargo.toml
```
opentelemetry = { version = "*", features = [ "rt-tokio" ] }
opentelemetry-otlp = { version = "*", features = [ "http-proto", "reqwest-client" ] }
tokio = { version = "*", features = ["full"] }
```

and yields:
```
OpenTelemetry trace error occurred. error sending request for url (https://api.honeycomb.io/v1/traces): error trying to connect: invalid URL, scheme is not http
```
How am I expected to configure this? Honeycomb needs HTTPS but it seems like how I'm using this crate is wrong.

**Comment by tobymurray** [comment, marked as answer]
I have no idea why, but the version of reqwest seems to be related to this. Relying on the reqwest client in opentelemetry-oltp seems to resolve this for me?

**Reply by maxmcd** [reply]
This is because `opentelemetry_otel` has [`reqwest = { version = "0.12", default-features = false }`](https://github.com/open-telemetry/opentelemetry-rust/blob/86dfd4b3db00ecb910d718ed367c672d7c586121/Cargo.toml#L39). 

If you add `reqwest` as a dependency directly it will [include the `default-tls` feature](https://github.com/seanmonstar/reqwest/blob/747737bd29096f34d516a20959aa7075b2a174e8/Cargo.toml#L37). 


Best bet is to change otlp to `opentelemetry-otlp = { version = "*", features = ["reqwest-rustls"] }` as @chinmoyrp is suggesting.

**Comment by chinmoyrp** [comment]
From what I understood this error is related to TLS. Adding the feature reqwest-rustls to opentelemetry-otlp fixed it for me.
